File: Barone.Client/src/app/modules/shared/models/compras.model.ts
import { Constructable } from "src/app/core/models/constructable";
import { InsumoModel } from "./insumo.model";
import { ProveedoresModel } from "./proveedor.model";

export class ComprasModel extends Constructable<Partial<ComprasModel>>{
    id: number;
    Cantidad: number;
    Precio: number;
    FechaCompra: number;
    Insumos_id: number;
    Proveedor_id: number;
    Insumo: InsumoModel;
    Proveedor: ProveedoresModel;

}

File: Barone.api/Models/ComprasModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Barone.api.Models
{
    public class ComprasModel
    {
        public int id { get; set; }
        public decimal Cantidad { get; set; }
        public decimal Precio { get; set; }
        public DateTime FechaCompra { get; set; }
        public virtual InsumoModel Insumo { get; set; }
        public virtual ProveedoresModel Proveedor { get; set; }

    }
}